lib/dal/src/func/authoring/execute.rs

```diff
@@ -0,0 +1,45 @@
+use telemetry::prelude::warn;
+
+use crate::func::authoring::{FuncAuthoringError, FuncAuthoringResult};
+use crate::func::FuncKind;
+use crate::{AttributePrototype, AttributeValue, DalContext, Func, FuncId};
+
+pub(crate) async fn execute_func(ctx: &DalContext, id: FuncId) -> FuncAuthoringResult<()> {
+    let func = Func::get_by_id_or_error(ctx, id).await?;
+
+    match func.kind {
+        FuncKind::Attribute => update_values_for_func(ctx, &func).await?,
+        FuncKind::Action => {
+            // TODO(nick): fully restore or wait for actions v2. Essentially, we need to run
+            // every prototype using the func id for every component.
+            warn!("skipping action execution...");
+            return Ok(());
+        }
+        kind => return Err(FuncAuthoringError::NotRunnable(id, kind)),
+    };
+
+    Ok(())
+}
+
+async fn update_values_for_func(ctx: &DalContext, func: &Func) -> FuncAuthoringResult<()> {
+    let attribute_prototype_ids = AttributePrototype::list_ids_for_func_id(ctx, func.id).await?;
+
+    if attribute_prototype_ids.is_empty() {
+        warn!(%func.id, "nothing to execute: no attribute prototype ids found for attribute func id");
+        return Ok(());
+    }
```

Comment:
This was an error in the past, but a no-op should suffice in case the prototypes fall out.